### PR TITLE
Image Extension

### PR DESCRIPTION
**Pull request by bharney0**
This is new markdig functionality that is being built for the Docs Platform to enhance images using the Markdig Extensions, which is being supported by the Docs Authoring Pack. The new image syntax
complex:
```
:::image type="complex" alt-text="Lorem Ipsum" source="./path":::

:::image-end
```
content:
```
:::image type="content" alt-text="Lorem Ipsum" source="./path":::
```
icon:
```
:::image type="icon" source="./path":::
```
- [x] Image type="content"
- [x] Image type="icon"
- [x] Image type="complex"
- [x] Add loc-scope to image
- [x] Preview support
- [x] Linting support
- [x] Auto-Completion

### CODE REVIEW

File: docs-markdown/markdownlint-custom-rules/image.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// @ts-check
+
```
Comment (adunndevster) [inline docs-markdown/markdownlint-custom-rules/image.js]: @bharney-msft  - Is this file just a stub? It is called image.js but it has xref rules?

Comment (adunndevster) [reply docs-markdown/markdownlint-custom-rules/image.js]: I'm just asking because I own the task to write image linting rules :)

Comment (bharney0) [reply docs-markdown/markdownlint-custom-rules/image.js]: Its just a stub that I was using for the task to mock out some sudo code.

Comment (adunndevster) [reply docs-markdown/markdownlint-custom-rules/image.js]: Got it.

**Pull request merged**